js/ui/marker.js

```diff
@@ -52,21 +54,22 @@ Marker.prototype = {
      * marker.remove();
      * @returns {Marker} `this`
      */
-    remove: function() {
+    remove: function () {
         if (this._map) {
             this._map.off('move', this._update);
             this._map = null;
         }
         var parent = this._el.parentNode;
         if (parent) parent.removeChild(this._el);
+        if (this._popup) this._closePopup();
```

Comment:
`this._closePopup` has an internal `if (this._popup)` guard -- either call it unguarded here, or add external guards everywhere it's called and remove the internal guard.